docs/usage/webhook/webhook.rst

```diff
@@ -0,0 +1,118 @@
+.. _usage_webhook:
+
+Redeploying on receving GitHub webhook
+======================================
+
+We'll move ``ffmpeg`` to a GitHub repo, and set up a webhook DataFlow such that whenever
+we push to the default branch, the new version is pulled and its docker container is built and run.
+
+Webhook Dataflow
+----------------
+
+We'll be using operations from ``dffml-operations-deploy`` and ``dffml-feature-git``.
+
+.. code-block:: console
+
+    $ pip install dffml-operations-deploy dffml-feature-git
+
+Setup a http server in ``ffmpeg/deploy/webhook``, to receive webhook and redploy ffmpeg
+
+.. code-block:: console
+
+    $ mkdir -p deploy/webhook/df deploy/webhook/mc/http
+    $ cat > /tmp/operations <<EOF
+    get_url_from_payload
+    clone_git_repo
+    check_if_default_branch
+    get_image_tag
+    get_running_containers
+    get_status_running_containers
+    parse_docker_commands
+    docker_build_image
+    restart_running_containers
+    cleanup_git_repo
+    EOF
+    $ dffml dataflow create -config yaml $(cat /tmp/operations) > deploy/webhook/df/webhook.yaml
+
+Config
+
+**deploy/webhook/mc/http/webhook.yaml**
+
+.. code-block:: console
+
+    $ cat > ./deploy/webhook/mc/http/webhook.yaml <<EOF
+    path: /webhook/github
+    output_mode: json
+    input_mode: json:git_payload
+    EOF
+
+Note that the input_mode is ``json:git_payload``, this means that inputs from post request will
+be parsed as JSON and then forwarded to dataflow as the ``git_payload`` definition..
+
+Deploy it in port 8081 as 8080 is being used by ffmpeg http service
+
+.. code-block:: console
+
+    $ dffml service http server -insecure -mc-config deploy/webhook -port 8081
+
+.. note::
+
+    If you're not setting this up on a server directly accessible on the internet,
+    here are two methods of exposing the webhook,
+
+    Using `localhost.run <https://localhost.run>`_
+
+    .. code-block:: console
+
+        $ ssh -R 80:localhost:8081 $RANDOM@ssh.localhost.run
+
+    Using ngrok
+
+    .. code-block:: console
+
+        $ ~/ngrok http 8081
+
+we'll be using ``ngrok`` in this tutorial
+
+    .. code-block:: console
+
+        $ ~/ngrok http 8081
+
+.. image:: ./images/ngrok_out.png
+
+Copy paste the output url to ``Payload URL`` in webhook settings of ffmpeg repo.
+
+.. image:: ./images/github_settings.png
```

Comment:
We need to open an issue to make sure that we support having this only happen with a secret.

We'll need to add some operation that verifies that secret == what's supposed to be. and we'll set it via a `@config` for the operation. And then make it output a bool which we can use as a condition.

Reply:
For now, it looks like the URL is off, replace with this:

![image](https://user-images.githubusercontent.com/5950433/81730962-256f6b00-9443-11ea-823d-2824ca92234b.png)

Reply:
The url was indeed off,Thanks!